lib/galaxy/security/validate_user_input.py

```diff
@@ -65,24 +67,27 @@ def validate_publicname(trans, publicname, user=None):
     return ''
 
 
-def transform_publicname(trans, publicname, user=None):
-    # User names must be at least four characters in length and contain only lower-case
-    # letters, numbers, and the '-' character.
+def transform_publicname(trans, publicname, user=None, check_dup=True):
+    """
+    Transform publicname to respect the minimum and maximum string length, and
+    the allowed characters.
+    FILL_CHAR is used to extend or replace characters.
+    """
     # TODO: Enhance to allow generation of semi-random publicnnames e.g., when valid but taken
     if user and user.username == publicname:
         return publicname
-    elif publicname not in ['None', None, '']:
+    if publicname not in ['None', None, '']:
         publicname = publicname.lower()
         publicname = re.sub(VALID_PUBLICNAME_SUB, FILL_CHAR, publicname)
         publicname = publicname.ljust(PUBLICNAME_MIN_LEN + 1, FILL_CHAR)[:PUBLICNAME_MAX_LEN]
```

Comment:
`FILL_CHAR` is set to '-' (minus) whereas our old clean function was using '_' (underscore) instead. This is valid but a little annoying for us. We will have two kind of pattern for galaxy public username. One with '_' and the other one with '-' depending of when the username has been generated (before or after galaxy 20.1) .  I think that as admin we can deal with it but for ours user it will be awkward.

Reply:
I guess you can use a database query to update the existing Galaxy usernames to use `-`?